CONTRIBUTING.md

```diff
@@ -188,11 +188,58 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subrepo)
 
-Currently building Rust will also build the following external projects:
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
 
 * [clippy](https://github.com/rust-lang/rust-clippy)
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies.
+
+If you want to synchronize or otherwise work with subrepos, install the `git subrepo` command via
+instructions found at https://github.com/ingydotnet/git-subrepo
+
+#### Creating a new subrepo dependency
+
+If you want to create a new subrepo dependency from an existing repository, call (from this
+repository's root directory!!)
+
+```
+git subrepo clone https://github.com/rust-lang/rust-clippy.git src/tools/clippy
+```
+
+This will create a new commit, which you may not rebase under any circumstances! Delete the commit
+and redo the operation if you need to rebase.
+
+Now you're done, the `src/tools/clippy` directory behaves as if clippy were part of the rustc
+monorepo, so no one but you (or others that synchronize subrepos) needs to have `git subrepo`
+installed.
+
+#### Synchronizing a subrepo
```

Comment:
This, too, is something that contributors *usually* shouldn't have to do, instead that would generally be tool maintainer's job?

Is there a way for us maintainers to be notified when our tools change in rustc so that we know we need to sync? (Longer-term, having a bot to also do the syncing would be even better of course. But some kind of notification is required, I think, to avoid silent divergence.)

Reply:
Contributors shouldn't care about this entire section. Where else do I put it?

We could probably extend the toolstate bot to check whether `git diff src/tools/clippy` has any output, and in that case ping the team